### PR TITLE
Added Typer.io Presence

### PR DESCRIPTION
**Pull request by dzlandis**
**Name of the service:**
Typer.io

**URL to website:**
https://typer.io

**Screenshots:**
Pic Pic Pic Pic Pic Pic Pic Pic Pic Pic Pic Pic Pic Pic Pic

### CODE REVIEW

**Comment by dzlandis** [comment]
Currently working on fixing Codacy Static Code Analysis issue 👍

**Comment by dzlandis** [comment]
Codacy Static Code Analysis finally approved 😀

File: websites/T/Typer.io/presence.ts [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+}),
+browsingStamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "logo",
+    startTimestamp: browsingStamp
+  };
+
+  if (document.location.hostname === "typer.io") {
```
Comment (ririxidev) [inline websites/T/Typer.io/presence.ts:12]: there is no need for this `if` statement.

File: websites/T/Typer.io/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    "tags": [
+      "typing",
+      "multiplayer",
+      "competition",
+      "typerio",
+      "race",
+      "typer",
+      "wpm",
+      "wordsperminute",
+      "words-per-minute"
```
Comment (ririxidev) [inline websites/T/Typer.io/dist/metadata.json:22]:
```suggestion
      "typing",
      "competition",
      "typerio",
      "race"
```

**Review by rxri — CHANGES_REQUESTED** [review]
and run prettier over presence file

**Comment by dzlandis** [comment]
The DeepScan appears to be failing because of issues on the Apple presence. Nothing I can do about that...

DeepScan Pic

**Comment by ririxidev** [comment]
I know, just ignore it lol

**Comment by dzlandis** [comment]
Let me know if you'd like me to revert the prettier stuff from this last commit: https://github.com/PreMiD/Presences/pull/3956/commits/f819f455300a180023afb4310bcbd998af629b2b
And revert it back to this one: https://github.com/PreMiD/Presences/pull/3956/commits/c689c99ad9bff6166b23f5c478262d8ce1c6e4f0

https://github.com/PreMiD/Presences/pull/3956/commits/c689c99ad9bff6166b23f5c478262d8ce1c6e4f0 was just fixing indent stuff and https://github.com/PreMiD/Presences/pull/3956/commits/f819f455300a180023afb4310bcbd998af629b2b was actually running the full prettier program.

Let me know which one you prefer. 👍

**Comment by ririxidev** [comment]
Well it doesn't look like running prettier because prettier would use tabs for example before `browsingStamp` variable so idk how you did it

**Comment by dzlandis** [comment]
> Well it doesn't look like running prettier because prettier would use tabs for example before `browsingStamp` variable so idk how you did it

I think it is possible that I messed things up when copying from https://prettier.io . I'll try it again and fix very soon.

**Review by rxri — DISMISSED** [review]

**Comment by dzlandis** [comment]
The developer of the website just let me know they added more hidden HTML elements for me to use so the time and race place stuff is accurate 100% of the time. Please don't merge until I make those changes to use them. I'll have  it done by later today. Thanks!

**Comment by dzlandis** [comment]
That was the last update. I am done and ready to be reviewed. 😀

**Review by rxri — DISMISSED** [review]

File: websites/T/Typer.io/presence.ts [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+        presenceData.details = `Placed ${quickplayRacePlace} in a quick play race!`;
+        presenceData.state = `${quickplayWPM}, ${quickplayAccuracy}, ${quickplayRaceTime}`;
+      }
+
+      if (
+        quickplayRacePlace === "-" &&
+        quickplayWPM !== "undefined" &&
+        quickplayRaceStatus === "Game has Ended"
+      ) {
+        presenceData.details = "The game ended, did not place.";
```
Comment (SlowLife1661) [inline websites/T/Typer.io/presence.ts:68]: Did not place? This sounds wrong to me

**Review by Slowlife01 — COMMENTED** [review]
the `.split("Jump to")` thing are they necessary?

File: websites/T/Typer.io/presence.ts (hunk 3 above)
Comment (dzlandis) [reply websites/T/Typer.io/presence.ts:68]: I can change it to "did not finish" if that makes more sense.

**Comment by dzlandis** [comment]
> the `.split("Jump to")` thing are they necessary?

I'm not 100% but I believe when I tested it, I couldn't get it to work without. I can try again but to my knowledge, it needed that for some reason. I could definitely be wrong though.

File: websites/T/Typer.io/presence.ts (hunk 3 above)
Comment (SlowLife1661) [reply websites/T/Typer.io/presence.ts:68]: yeah well, the context is that they didn't finish the game, is that right

Comment (dzlandis) [reply websites/T/Typer.io/presence.ts:68]: Yep. Fixing.

**Comment by dzlandis** [comment]
Should be ready for another check! 👍🏻

File: websites/T/Typer.io/presence.ts [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+    try {
+      const user = document
+        .querySelector(
+          "#__next > div.Profile_root__2QIUs > div.Profile_headerContainer__IYvIA > div.Profile_profileContainer__k2Fu9 > div.Profile_wrapper__3Ghk7 > div.Profile_content__rWK4h > h3"
+        )
+        .textContent.trim();
+      presenceData.details = "Viewing a users profile:";
+      presenceData.state = user;
+    } catch {
+      presenceData.details = "Viewing a users profile.";
```
Comment (SlowLife1661) [inline websites/T/Typer.io/presence.ts:200]: You mean `user`

Comment (ririxidev) [reply websites/T/Typer.io/presence.ts:200]:
```suggestion
      presenceData.details = `Viewing a ${user}'s profile:`;
    } catch {
      presenceData.details = "Viewing a user profile.";
```

**Review by Slowlife01 — APPROVED** [review]

**Review by rxri — APPROVED** [review]

**Pull request merged**